### DISCUSSION TITLE
Can't get homebrew installed

### DISCUSSION THREAD

**Discussion by skibum12** [Homebrew/discussions#1218] [Getting started]
I must be missing something obvious but google doesn't find it.

Installing on a Mac running Big Sur 11.2.3.

I get the following failure:

==> Downloading and installing Homebrew...

fatal: Unable to find remote helper for 'https'

Failed during: git fetch --force origin

**Comment by gromgit** [comment]
Try running `xcode-select --install` to instal the Command Line Tools, then run the Homebrew installation step again.

**Reply by skibum12** [reply]
xcode-select: error: command line tools are already installed, use "Software Update" to install updates

**Reply by carlocab** [reply]
Try doing
```
sudo rm -rf /Library/Developer/CommandLineTools
xcode-select --install
```
The `sudo` command will ask for your password. (Please be very careful when doing `sudo rm -rf`.)

**Reply by skibum12** [reply]
Tried that.  Removed them, reinstalled them.  Then tried installing again, same error as above.

**Reply by carlocab** [reply]
Does this work:
```
cd /tmp
git clone https://github.com/Homebrew/brew.git
```
This should create a directory in `/tmp` called `brew`. Feel free to `rm -rf /tmp/brew` afterward -- no need for `sudo`. (Though `/tmp` is cleaned up on restart, so you can also just leave it there.)

**Reply by skibum12** [reply]
% git clone https://github.com/Homebrew/brew.git
Cloning into 'brew'...
fatal: Unable to find remote helper for 'https'

**Reply by carlocab** [reply]
There seems to be a problem with your system `git`.

What is the output of `which -a git`?

**Reply by skibum12** [reply]
```
% which -a git
/usr/local/bin/git
/usr/bin/git
```

**Reply by SMillerDev** [reply]
What does `ls -al /usr/local/bin/git` say?

**Reply by skibum12** [reply]
```
% ls -al /usr/local/bin/git
lrwxr-xr-x  1 root  wheel  14 Dec 26  2017 /usr/local/bin/git -> ../git/bin/git
```

**Reply by carlocab** [reply]
You seem to have a pre-existing install of `git` that's broken. Does this work:
```
cd /tmp
/usr/bin/git clone https://github.com/Homebrew/brew.git
```

**Reply by skibum12** [reply]
Yes that seems to work.   I don't recall ever installing git on this machine.

```
% /usr/bin/git clone https://github.com/Homebrew/brew.git
Cloning into 'brew'...
remote: Enumerating objects: 127, done.
remote: Counting objects: 100% (127/127), done.
remote: Compressing objects: 100% (82/82), done.
remote: Total 179414 (delta 68), reused 85 (delta 45), pack-reused 179287
Receiving objects: 100% (179414/179414), 47.14 MiB | 13.37 MiB/s, done.
Resolving deltas: 100% (133116/133116), done.
```

**Reply by carlocab** [reply]
Looks like the the `git` at `/usr/local/bin/git` is broken. I'd try to uninstall that. What's the output of
```
ls -al /usr/local/git
ls -al /usr/local/git/bin
```

**Reply by skibum12** [reply]
```
ls -al /usr/local/git
total 8
drwxr-xr-x  10 root  wheel  320 Mar 19  2019 .
drwxr-xr-x  17 root  wheel  544 Apr  6 17:17 ..
drwxr-xr-x   3 root  wheel   96 Nov  8  2017 Library
drwxr-xr-x  10 root  wheel  320 Dec 26  2017 bin
drwxr-xr-x   3 root  wheel   96 Nov  8  2017 contrib
drwxr-xr-x   3 root  wheel   96 Nov  8  2017 etc
drwxr-xr-x   3 root  wheel   96 Nov  8  2017 lib
drwxr-xr-x   3 root  wheel   96 Nov  8  2017 libexec
drwxr-xr-x   7 root  wheel  224 Nov  8  2017 share
-rwxr-xr-x   1 root  wheel  718 Nov  8  2017 uninstall.sh

ls -al /usr/local/git/bin
total 25344
drwxr-xr-x  10 root  wheel      320 Dec 26  2017 .
drwxr-xr-x  10 root  wheel      320 Mar 19  2019 ..
-rwxr-xr-x   1 root  wheel  5941024 Nov  8  2017 git
-rwxr-xr-x   1 root  wheel    26660 Nov  8  2017 git-credential-osxkeychain
-rwxr-xr-x   2 root  wheel   162748 Nov  8  2017 git-cvsserver
lrwxr-xr-x   1 root  wheel        3 Dec 26  2017 git-receive-pack -> git
-rwxr-xr-x   2 root  wheel  3138676 Nov  8  2017 git-shell
lrwxr-xr-x   1 root  wheel        3 Dec 26  2017 git-upload-archive -> git
-rwxr-xr-x   2 root  wheel  3346084 Nov  8  2017 git-upload-pack
-rwxr-xr-x   1 root  wheel   351673 Nov  8  2017 gitk
```

**Reply by carlocab** [reply]
Running
```
/usr/local/git/uninstall.sh
```
looks like what you want here. I'd have a look inside that script to make sure it's not doing anything weird (considering that you don't recall ever installing `git` here). Assuming it looks fine, I'd run it. You will almost certainly need `sudo` for it to work.

**Reply by skibum12** [reply]
Awesome.  Removing that git version allowed homebrew to install.  Thanks for everyone's help!
